go/vt/vtctl/query.go

```diff
@@ -526,23 +526,20 @@ func commandVtTabletStreamHealth(ctx context.Context, wr *wrangler.Wrangler, sub
 		return fmt.Errorf("cannot connect to tablet %v: %v", tabletAlias, err)
 	}
 
-	stream, err := conn.StreamHealth(ctx)
-	if err != nil {
-		return err
-	}
-	for i := 0; i < *count; i++ {
-		shr, err := stream.Recv()
-		if err != nil {
-			return fmt.Errorf("stream ended early: %v", err)
-		}
+	i := 0
+	return conn.StreamHealth(ctx, func(shr *querypb.StreamHealthResponse) error {
```

Comment:
This one might be missing the case where it returns nil, but we haven't gotten enough responses. i.e. we ask for 'count', but the server returns nil with i < count.